### DISCUSSION TITLE
[QUESTION] Export Huawei Health

### DISCUSSION THREAD

**Discussion by Kooch22** [CTHRU/Huawei-TCX-Converter#25] [Show and tell]
It's not a Hitrava bug, but a Huawei Health download bug.

Normally I would get the downloadfile within 24 hours. Last time it took 6 days, and then the file was empty.

Anyone else having this isseu?

Kooch

**Comment by RobinTH90** [comment]
Got the same problem. Even the 24h wait annoys me. As an iPhone user, I switched to a rooted Android Emulator and pull out the files manually.

**Comment by mbraeutig** [comment]
I have the same poblem,
it took 6 days, and then the file was empty.

**Comment by mbraeutig** [comment]
I have make a report here:
https://consumer.huawei.com/en/legal/privacy-questions/

**Comment by lamerjack** [comment]
Considering that backup metod is not working and the email method seems to fail.. im considering to copy files in data/com. huawei. health/files/ 
I dont own yet a GT2 i want to be sure that i will be able to export data... so on a rooted phone i did an activity just using the phone...
But i didnt found hitrack files in folder data/com. huawei. health/files/ 
Only smartwhatches create these files?
When these files are created?
Are created only when i export data from smartwhach to the phone?
Or if i export to smartwhatch to phone A, then i will see Hitrack files also on phone B where there is Huawei healt app with the same account and the activities are syncronized?

**Comment by RobinTH90** [comment]
I'm not sure about the "phone only" thing but this is how I do it: I use my GT2 with phone A and track my activities via GT2. I sync everything to phone A and enable cloud sync. On phone B (rooted android emulator) I sync all the tracked data via cloud sync. I found the hitrack files data/com.huawei.health/files/. But sometimes there were no hitrack files so I figured out that the files get deleted. I can generate the file again when I open the activity in Huawei Health on phone B to see the GPS track. Than the hitrack file gets generated and I can export it via any file manager.

**Comment by lamerjack** [comment]
I was planning to use an orange pi (raspberry clone) with android... because my phone B doesn't boot anymore....
But android emulator sound good.
Wich one do you suggest me?

**Comment by RobinTH90** [comment]
I think an emulator which has root built in will do it. I used Nox before but a few days ago it won't cloud sync anymore. Don't know why. Than I changed to MEmu and it works great so far.

**Comment by lamerjack** [comment]
Just tested MEmu works fine!
Virtualbox can mount disk image of MEmu so is possible to do a script... just i dont know if Virtualbox and MEmu can access the disk image at the same time!

**Comment by lamerjack** [comment]
Hi Robin,
Maybe this will semplify your work:
If you install terminal for android
you can do these commands in terminal:
cd /data/data/
tar -cvf /storage/sdcard0/Download/com.huawei.health.tar com.huawei.health
and you fill find a tar file in the shared folder of memu

**Comment by lamerjack** [comment]
this 
tar -cvf /storage/sdcard0/Download/com.huawei.health.tar com.huawei.health/files/HiTrack_*
is more optimized and copy only hitrackfiles

**Comment by RobinTH90** [comment]
now you are going crazy :D but works great. thanks.

**Comment by lamerjack** [comment]
Going Crazy?
Look at this:

install termux app in Memu
Open termux and write these commands:

apt install python git
pip install xmlschema

git clone https://gitlab.com/st42/termux-sudo
cd termux-sudo
cat sudo > /data/data/com.termux/files/usr/bin/sudo
chmod 700 /data/data/com.termux/files/usr/bin/sudo

cd
git clone https://github.com/CTHRU/Hitrava
chmod 755 Hitrava/Hitrava.py

nano export.sh
#Now paste these commands insite nano 
if (( $# != 1 ))
then
  set -- "1980-01-01"
fi
myvar="$PWD"
cd /data/data/
tar -cvf "$myvar"/com.huawei.health.tar com.huawei.health/files/HiTrack_*
cd "$myvar"
python "$myvar"/Hitrava/Hitrava.py --tar com.huawei.health.tar --from_date $1 --output_dir /storage/sdcard0/Download/Strava

#and save with ctrl x

chmod755 export.sh

once done every time that you want to export tcx open termux and write:
./export.sh
or
./export.sh 2020-09-16

you will find a Strava folder in the shared download folder the with all tcx files.
p.s. i don't like strava but these files works fine also with Runkeeper.

**Comment by RobinTH90** [comment]
wow thats impressive. works great so far. saves me many steps I had to do before. But now I'm thinking what to do when Huawei Health is not creating those HiTrack files. Like I stated before sometimes I cloud sync to the emulator and Health didn't create any new HiTrack file. I had to access the activity in Health and it creates the file but also sometimes deletes it immediately when I leave the activity ? But with your script its faster anyway :D thanks

**Comment by smonov** [comment]
>  Kooch22 wrote:
>"It's not a Hitrava bug, but a Huawei Health download bug.
>Normally I would get the downloadfile within 24 hours. Last time it took 6 days, and then the file was empty.
>Anyone else having this issue?"
 ===================================================

Same for me I have recieved 3 or 4 empty files in a row, in September. After that exports are running well well again.
Could it be because of export request limits in Huawei Helth page? Does anybody knows how many times you can export the data during the current month?

**Comment by thgian** [comment]
Hi all,
I have experienced the same problem. After numerous requests, I was receiving an empty file. I made several complaints to the official Huawei representative and they were advising me to delete the Health app and reinstall it (!!!!!!). 
I never did it and some days later (1st of Oct) I received an email from Aspiegel company (Huawei's subsidiary) saying that the problem was from the data size limitations of the "gateway Nginx". They increased the limit of the data that can be downloaded (more than 1M). After this change, all came back to normality. 
Hope that helps.
;-)

**Comment by lperozzi** [comment]
Hi all and thanks for this thread,

I tried to follow the @lamerjack workflow and when I run `./export.sh`  I have the following.

`tar: com.huawei.health/files: Cannot stat: Permission denied`

I've tried to `chmod` it without success.

I just installed Termux together with Health app in MEmu. Maybe I forget some steps? Many thanks or your help

**Reply by rftg1996** [reply]
same

**Reply by rftg1996** [reply]
did u solve? on a german blog i read that with "sudo ./export.sh" it should work. In fact the terminal accepts the command but i can't find the file

**Comment by albynos83** [comment]
Hi! The extraction went well. I recovered 75 activities which were converted and then transferred to Strava. But impossible to recover the last month (December). Still, the December list is there. Numbers, dates, scores, etc. Everything is there but in gray color. Inselectable. Did you have a similar problem?

**Reply by CTHRU** [reply]
@albynos83 I assume you are referring to the web version. What you noticed, is 'normal'. It's currently an alpha version and I decided to release it in trial mode only for now. This means that some of the last activities will be greyed out. It's also mentioned in the alpha software notice [here](https://cthru.hopto.org/hitrava-web-alpha/).
I'm trying to continue to work on it in my spare time, but due to personal reasons, I am currently unable to spend the necessary time. I thank you for your patience.
See also #32.

**Reply by albynos83** [reply]
I thought it was my fault. Thank you for your explanations! And congratulations on this hard work, I hope you will come to the end of your ideas!
